app/controllers/comments_controller.rb

```diff
@@ -16,7 +16,7 @@ def create
     @post.screenshots = params[:post][:screenshots]
 
     if @post.save
-      redirect_to doc_path(@doc)
+      redirect_to path_based_on_category_public_status(@doc)
```

Comment:
I think this might be better done by passing a source_of_comment attr instead of looking purely at whether or not it is public.  I was testing it and was surprised to get redirected from the admin experience back out to the public layout after making a comment